core/modules/replica/IngestServerConnection.cc

```diff
@@ -404,93 +405,105 @@ void IngestServerConnection::_loadDataIntoTable() {
+                    ""
+                )
+            )
+        );
 
-        string const sqlDatabase = conn->sqlId(_databaseInfo.name);
-        string const sqlPartition = conn->sqlPartitionId(_transactionId);
+        string const sqlDatabase = h.conn->sqlId(_databaseInfo.name);
+        string const sqlPartition = h.conn->sqlPartitionId(_transactionId);
 
         vector<string> statements;
 
         if (_isPartitioned) {
             
-            // Chunked tables are created from the prototype table which is expected
-            // to exist in the database before attempting data loading.
-
-            string const sqlProtoTable = sqlDatabase + "." + conn->sqlId(_table);
-            string const sqlTable = sqlDatabase + "." + conn->sqlId(_table + "_" + to_string(_chunk));
-            string const sqlFullOverlapTable  =
-                sqlDatabase + "." + conn->sqlId(_table + "FullOverlap_" + to_string(_chunk));
-
-            string const sqlDummyChunkTable =
-                sqlDatabase + "." + conn->sqlId(_table + "_" + to_string(lsst::qserv::DUMMY_CHUNK));
-
-            string const sqlOverlapDummyChunkTable =
-                sqlDatabase + "." + conn->sqlId(_table + "FullOverlap_" + to_string(lsst::qserv::DUMMY_CHUNK));
-
-            vector<string> const tablesToBeCreated = {
-                sqlTable,
-                sqlFullOverlapTable,
-                sqlDummyChunkTable,
-                sqlOverlapDummyChunkTable
-            };
-            for (auto&& table: tablesToBeCreated) {
-                statements.push_back(
-                    "CREATE TABLE IF NOT EXISTS " + table + " LIKE " + sqlProtoTable
-                );
-                statements.push_back(
-                    "ALTER TABLE " + table + " ADD PARTITION IF NOT EXISTS (PARTITION " + sqlPartition +
-                        " VALUES IN (" + to_string(_transactionId) + "))"
-                );
+            // Note, that the algorithm will create chunked tables for _ALL_ partitioned
+            // tables (not just for the current one) to ensure they have representations
+            // in all chunks touched by the ingest workflows. Missing representations would
+            // case Qserv to fail when processing queries involving these tables.
+
+            for (auto&& table: _databaseInfo.partitionedTables) {
+
+                // Chunked tables are created from the prototype table which is expected
+                // to exist in the database before attempting data loading.
+
+                bool const overlap = true;
+                string const sqlProtoTable       = sqlDatabase + "." + h.conn->sqlId(table);
+                string const sqlTable            = sqlDatabase + "." + h.conn->sqlId(ChunkedTable(table, _chunk, not overlap).name());
+                string const sqlFullOverlapTable = sqlDatabase + "." + h.conn->sqlId(ChunkedTable(table, _chunk, overlap).name());
+
+                vector<string> const tablesToBeCreated = {
```

Comment:
Could do just `string tablesToBeCreated[]`, no need for vector.

Reply:
This optimization will the next block of code to fail:
```
                for (auto&& table: tablesToBeCreated) {
                    statements.push_back(
                        "CREATE TABLE IF NOT EXISTS " + table + " LIKE " + sqlProtoTable
                    );
                    statements.push_back(
                        "ALTER TABLE " + table + " ADD PARTITION IF NOT EXISTS (PARTITION " + sqlPartition +
                            " VALUES IN (" + to_string(_transactionId) + "))"
                    );
                }
```
It would require wrapping table names into `string(table)` to successfully concatenate strings.

Reply:
I don't think so, `table` would still be a `std::string` (or reference to it).

Reply:
You're right! Fixed.